ServerCore/Pages/Submissions/Index.cshtml

```diff
@@ -700,20 +698,43 @@ else
                     // Response Code enum is in Helpers/SubmissionEvaluator
                     const messageArea = document.querySelector("#initialMessage");
                     let updateVal = "";
+                    let responseString = (responseData.completeResponse ?? "") + (responseData.freeformResponse ?? "");
+                    if (responseString.startsWith("Html.Raw")) { 
+                        responseString = responseString.substring(9, responseString.length - 1);
```

Comment:
I assume you want to take the whole response string but remove the first "Html.Raw" bit.

If so, don't you want the below since the second parameter of the substring method is the length of the substring?
int lenToRemove = "Html.Raw".Length;
int substringLen = responseString.Length - lenToRemove;
responseString.substring(lenToRemove, substringLen)

Reply:
HtmlRawHelper.Display() is better, it addresses other cases as well.

Reply:
@digit01Wave this is Javascript's substring, which uses endIndex instead of length as the second parameter.  substr is the one that uses length, and it's deprecated per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr

Reply:
Ohh. Makes sense. My bad. My mind was stuck in C# land